cdap-spark-core/src/main/scala/co/cask/cdap/app/runtime/spark/BatchReadableRDD.scala

```diff
@@ -37,6 +37,7 @@ import scala.reflect.ClassTag
   */
 class BatchReadableRDD[K: ClassTag, V: ClassTag](@transient sc: SparkContext,
                                                  @transient batchReadable: BatchReadable[K, V],
+                                                 datasetNamespace: Option[String],
```

Comment:
Oh forgive me. I found the way to do these things more elegantly. As we know the namespace actually as soon as user call the API in sec. Will change it and commit again.